collector/consumer/processor/k8sprocessor/kubernetes_processor.go

```diff
@@ -85,15 +86,36 @@ func (p *K8sMetadataProcessor) processTcpMetric(dataGroup *model.DataGroup) {
 func (p *K8sMetadataProcessor) addK8sMetaDataForClientLabel(labelMap *model.AttributeMap) {
 	// add metadata for src
 	containerId := labelMap.GetStringValue(constlabels.ContainerId)
-	labelMap.UpdateAddStringValue(constlabels.SrcContainerId, containerId)
-	resInfo, ok := p.metadata.GetByContainerId(containerId)
-	if ok {
-		addContainerMetaInfoLabelSRC(labelMap, resInfo)
+	if containerId != "" {
+		labelMap.UpdateAddStringValue(constlabels.SrcContainerId, containerId)
+		resInfo, ok := p.metadata.GetByContainerId(containerId)
+		if ok {
+			addContainerMetaInfoLabelSRC(labelMap, resInfo)
+		} else {
+			labelMap.UpdateAddStringValue(constlabels.SrcNodeIp, p.localNodeIp)
+			labelMap.UpdateAddStringValue(constlabels.SrcNode, p.localNodeName)
+			labelMap.UpdateAddStringValue(constlabels.SrcNamespace, constlabels.InternalClusterNamespace)
+		}
 	} else {
-		labelMap.UpdateAddStringValue(constlabels.SrcNodeIp, p.localNodeIp)
-		labelMap.UpdateAddStringValue(constlabels.SrcNode, p.localNodeName)
-		labelMap.UpdateAddStringValue(constlabels.SrcNamespace, constlabels.InternalClusterNamespace)
+		srcIp := labelMap.GetStringValue(constlabels.SrcIp)
+		if srcIp == loopbackIp {
+			labelMap.UpdateAddStringValue(constlabels.SrcNodeIp, p.localNodeIp)
+			labelMap.UpdateAddStringValue(constlabels.SrcNode, p.localNodeName)
+		}
```

Comment:
LGTM, Although The k8s information searched may be incorrect when processing data from the loopback device.